src/main/java/com/wcc/platform/service/EventService.java

```diff
@@ -22,14 +28,32 @@ public EventService(final ObjectMapper objectMapper) {
   }
 
   /**
-   * Read Json and convert to POJO event page.
+   * Read Json and convert to paginated event page.
    *
    * @return POJO eventsPage
    */
-  public EventsPage getEvents() {
+  public EventsPage getEvents(final int currentPage, final int pageSize) {
     try {
       final var data = FileUtil.readFileAsString(EVENTS.getFileName());
-      return objectMapper.readValue(data, EventsPage.class);
+
+      final var page = objectMapper.readValue(data, EventsPage.class);
+      final var allEvents = page.data().items();
+
+      final List<Event> paginatedEvents =
+          PaginationUtil.getPaginatedResult(allEvents, currentPage, pageSize);
+      final Pagination paginationRecord =
+          new Pagination(
+              allEvents.size(),
+              PaginationUtil.getTotalPages(allEvents, pageSize),
+              currentPage,
+              pageSize);
+      final PageData<Event> eventPageData = new PageData<>(null, null, null, paginatedEvents);
```

Comment:
I guess you could create another contractor for PageData without the first 3 params, what do you think?

Reply:
Removed other params for now